pyccel/ast/numpyext.py

```diff
@@ -583,6 +557,10 @@ def __new__(cls, shape, dtype='float', order='C'):
         order = cls._process_order(order)
 
         return Basic.__new__(cls, shape, dtype, order, precision)
+
+    def __init__(self, shape, dtype=None, order='C'):
+        NumpyFull.__init__(self, shape, None, dtype, order)
```

Comment:
Before this addition, I suppose that the `NumpyFull.__init__` was called from the constructors of `NumpyEmpty`, `NumpyZeros`, `NumpyOnes` with the wrong arguments... How come that this did not result in some errors?

Reply:
`NumpyFull` didn't have an `__init__` function before. So it was the sympy constructor which was called with `*args, **kwargs`

Reply:
Right! Thanks